src/features/campaigntask/index.js

```diff
@@ -128,79 +153,55 @@ function AdequacyTask(props) {
   return <RatingTask description={TASK_DESCRIPTIONS.adequacy} {...props} />;
 }
 
-const TASK_POOL = [
-  {
-    type: "comparison",
-    source: "Who let the dogs out?",
-    target: "Hver lét hundana út?",
-    target2: "Hver hleypti hundunum út?",
-  },
-  {
-    type: "fluidity",
-    source: "Who let the dogs out?",
-    target: "Hver lét hundana út?",
-  },
-  {
-    type: "adequacy",
-    source: "Who let the dogs out?",
-    target: "Hver lét hundana út?",
-  },
-  {
-    type: "comparison",
-    source: "Who let the dogs out?",
-    target: "Hver lét hundana út?",
-    target2: "Hver hleypti hundunum út?",
-  },
-  {
-    type: "fluidity",
-    source: "Who let the dogs out?",
-    target: "Hver lét hundana út?",
-  },
-  {
-    type: "adequacy",
-    source: "Who let the dogs out?",
-    target: "Hver lét hundana út?",
-  },
-];
-
 function CampaignTask() {
-  const [poolIdx, setPoolIdx] = useState(0);
-  const curTask = TASK_POOL[poolIdx];
-  const tasksLeft = TASK_POOL.length - poolIdx;
-  const progress = Math.floor(
-    (100 * (TASK_POOL.length - tasksLeft)) / TASK_POOL.length
-  );
+  const [tasksLeft, setTasksLeft] = useState(30);
+  const { id, mode } = useParams();
+  const progress = Math.floor(100 * ((30 - tasksLeft) / 30));
+
+  const [task, setTask] = useState(null);
+
+  function updateTask() {
+    getTask(id, mode).then((response) => setTask(response.data));
+  }
+
+  useEffect(() => {
+    updateTask();
+  }, []);
+
+  if (task === null) {
+    return <></>;
+  }
+
+  if (tasksLeft < 1) {
+    return <Redirect to="/campaigns" />;
+  }
+
+  function answerAndGetNext(answerData) {
+    setTasksLeft(tasksLeft - 1);
+    answerTask(id, answerData).then(() => updateTask());
```

Comment:
You probably want to have some error handling here and decrement the tasks left in the positive case.